internal/components/monitor/helpers.go

```diff
@@ -0,0 +1,45 @@
+package monitor
+
+import (
+	"fmt"
+	"slices"
+)
+
+// ValidateMonitoringOperation checks if the monitoring operation is supported
+func ValidateMonitoringOperation(operation string) bool {
+	supportedOps := []string{
+		string(OpMetrics), string(OpResourceHealth), string(OpAppInsights),
+		string(OpDiagnostics), string(OpControlPlaneLogs),
+	}
+	return slices.Contains(supportedOps, operation)
+}
+
+// GetSupportedMonitoringOperations returns all supported monitoring operations
+func GetSupportedMonitoringOperations() []string {
+	return []string{
+		string(OpMetrics), string(OpResourceHealth), string(OpAppInsights),
+		string(OpDiagnostics), string(OpControlPlaneLogs),
+	}
+}
+
+// ValidateMetricsQueryType checks if the metrics query type is supported
+func ValidateMetricsQueryType(queryType string) bool {
+	supportedTypes := []string{"list", "list-definitions", "list-namespaces"}
+	return slices.Contains(supportedTypes, queryType)
+}
+
+// MapMetricsQueryTypeToCommand maps a metrics query type to its corresponding az command
+func MapMetricsQueryTypeToCommand(queryType string) (string, error) {
+	commandMap := map[string]string{
+		"list":             "az monitor metrics list",
+		"list-definitions": "az monitor metrics list-definitions",
+		"list-namespaces":  "az monitor metrics list-namespaces",
+	}
+
+	cmd, exists := commandMap[queryType]
+	if !exists {
```

Comment:
The error message could be more helpful by suggesting valid alternatives:

```go
return "", fmt.Errorf("unsupported metrics query type '%s'. Supported types: list, list-definitions, list-namespaces", queryType)
```

This provides actionable information to help users correct their input.